app/assets/javascripts/discourse/app/components/discourse-root.js

```diff
@@ -0,0 +1,21 @@
+import Component from "@ember/component";
+
+let componentArgs = { tagName: "div", className: "discourse-root" };
```

Comment:
I didn't try this but I'm surprised it works when tagName is replaced by `""` what is the component doing from className? it just ignores the fact that it can't do anything with it, I thought it would error. Maybe it only error for bindings.

Reply:
It's not replaced in this case - the component is built either one way or the other, depending on features.

Reply:
It's not replaced in this case, the component class is built one way or the other depending on the features.